src/core/model/XojPage.h

```diff
@@ -56,17 +56,16 @@ class XojPage: public PageHandler {
 
     size_t getPdfPageNr() const;
 
-    bool isAnnotated();
+    bool isAnnotated() const;
 
     void setBackgroundColor(Color color);
     Color getBackgroundColor() const;
 
     std::vector<Layer*>* getLayers();
-    size_t getLayerCount();
-    int getSelectedLayerId();
-    void setSelectedLayerId(int id);
-    static bool isLayerVisible(Layer* layer);
-    bool isLayerVisible(int layerId);
+    size_t getLayerCount() const;
+    size_t getSelectedLayerId();
```

Comment:
Does this getter modify something? Or can it be const.

Reply:
Looks like `getSelectedLayerId` will insert a layer if there are no layers in the page. That's pretty undesirable imo, we should refactor it (perhaps in a followup) so that the page layers are always in a valid state.

Reply:
I agree there is something here. For a follow up PR.